Etcd 2.3.3 proxy panic - CloseNotify called after ServeHTTP finished

An etcd instance running 2.3.3 on linux has panicked this morning. The node was a proxy node and it failed with the following trace:

```
panic: net/http: CloseNotify called after ServeHTTP finished

goroutine 35455 [running]:
panic(0xbeaea0, 0xc8204262b0)
        /usr/local/go/src/runtime/panic.go:481 +0x3e6
net/http.(*response).CloseNotify(0xc8200721a0, 0x0)
        /usr/local/go/src/net/http/server.go:1535 +0x9d
github.com/coreos/etcd/proxy.(*reverseProxy).ServeHTTP.func1(0x7f623aa76f48, 0xc8200721a0, 0xc820444008, 0xc820197a40, 0xc820444160, 0xc8207de070)
        /home/gyuho/go/src/github.com/coreos/etcd/release/etcd/gopath/src/github.com/coreos/etcd/proxy/reverse.go:116 +0x39
created by github.com/coreos/etcd/proxy.(*reverseProxy).ServeHTTP
        /home/gyuho/go/src/github.com/coreos/etcd/release/etcd/gopath/src/github.com/coreos/etcd/proxy/reverse.go:122 +0xa1d
```

No special operation (cluster has been running for a while without any problem)

Do you have any hint?

Thanks a lot,

--- a/proxy/httpproxy/reverse.go
+++ b/proxy/httpproxy/reverse.go
@@ -112,9 +112,10 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 	closeNotifier, ok := rw.(http.CloseNotifier)
 	cancel := httputil.RequestCanceler(p.transport, proxyreq)
 	if ok {
+		closeCh := closeNotifier.CloseNotify()
 		go func() {
 			select {
-			case <-closeNotifier.CloseNotify():
+			case <-closeCh:
 				atomic.StoreInt32(&requestClosed, 1)
 				log.Printf("proxy: client %v closed request prematurely", clientreq.RemoteAddr)
 				cancel()

